### ISSUE TITLE
Transpiling `qss.AceCR` returns garbage

### ISSUE THREAD

**Issue by dowusu-antwi** [SupertechLabs/qiskit-superstaq#274]
EDIT: see comments below; this is actually giving us what we expect, from transpiling our definition in https://github.com/SupertechLabs/qiskit-superstaq/blob/9f544e285ac89147721683a733c86aa726600eaa/qiskit_superstaq/custom_gates.py#L41 But it may bring up a related issue.

---

### What is happening?

@knsmith discovered that transpiling `AceCR` gives unexpected results.

### How can we reproduce the issue?

```python
import qiskit_superstaq as qss
qc = qiskit.QuantumCircuit(2)
qc.append(qss.AceCR("+-"), [0, 1])
qc.draw()
     ┌───────────┐
q_0: ┤0          ├
     │  Acecr_pm │
q_1: ┤1          ├
     └───────────┘

qiskit.transpile(qc, jakarta).draw(fold=-1)
global phase: π/2
                                                                ┌───┐                                                  
      q_0 -> 0 ──────────────────────────────■───────────────■──┤ X ├──■────────────────■──────────────────────────────
               ┌─────────┐┌────┐┌─────────┐┌─┴─┐┌─────────┐┌─┴─┐└───┘┌─┴─┐┌──────────┐┌─┴─┐┌─────────┐┌────┐┌─────────┐
      q_1 -> 1 ┤ Rz(π/2) ├┤ √X ├┤ Rz(π/2) ├┤ X ├┤ Rz(π/4) ├┤ X ├─────┤ X ├┤ Rz(-π/4) ├┤ X ├┤ Rz(π/2) ├┤ √X ├┤ Rz(π/2) ├
               └─────────┘└────┘└─────────┘└───┘└─────────┘└───┘     └───┘└──────────┘└───┘└─────────┘└────┘└─────────┘
ancilla_0 -> 2 ────────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                                       
ancilla_1 -> 3 ────────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                                       
ancilla_2 -> 4 ────────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                                       
ancilla_3 -> 5 ────────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                                       
ancilla_4 -> 6 ────────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                                       
```

### What should happen?

Transpiling should fail (return an error), or do nothing:
```python
>>>qc.draw()
     ┌───────────┐
q_0: ┤0          ├
     │  Acecr_pm │
q_1: ┤1          ├
     └───────────┘

>>>qiskit.transpile(qc, jakarta).draw(fold=-1)
     ┌───────────┐
q_0: ┤0          ├
     │  Acecr_pm │
q_1: ┤1          ├
     └───────────┘
```

### Environment

- **SuperstaQ version**: 0.3.6
- **general-superstaq version**: 0.3.2
- **cirq-superstaq version**: 0.3.6
- **Operating system**: Ubuntu 22.04
- **Python version**: 3.8
- **Any other relevant dependencies**: None

**Comment by dowusu-antwi** [comment]
@richrines1 (from Pranav, possibly related is the gate definition [here](https://github.com/SupertechLabs/qiskit-superstaq/blob/9f544e285ac89147721683a733c86aa726600eaa/qiskit_superstaq/custom_gates.py#L41) -- maybe it's transpiling that? But there are CNOTs in it, which would be weird.):

**Comment by richrines1** [comment]
that's right. qiskit.transpile() will keep decomposing gates (by replacing them with their `.definition`) until the circuit can be expressed in the backend's basis gates, which in this case are:
```python
>>> jakarta.configuration().basis_gates
['id', 'rz', 'sx', 'x', 'cx', 'reset']
```
so in this case it's expanding these definitions:
```python
>>> print(qss.AceCR("+-").definition)
     ┌───────────┐┌───┐┌────────────┐
q_0: ┤0          ├┤ X ├┤0           ├
     │  Rzx(π/4) │└───┘│  Rzx(-π/4) │
q_1: ┤1          ├─────┤1           ├
     └───────────┘     └────────────┘
>>> print(qiskit.circuit.library.RZXGate(np.pi / 4).definition
q_0: ───────■───────────────■───────
     ┌───┐┌─┴─┐┌─────────┐┌─┴─┐┌───┐
q_1: ┤ H ├┤ X ├┤ Rz(π/4) ├┤ X ├┤ H ├
     └───┘└───┘└─────────┘└───┘└───┘
```

**Comment by dowusu-antwi** [comment]
Yep, just found that out on my end too. But Kate's tomography results shows that this decomposition significantly reduces fidelity, which might still be relevant.

**Comment by richrines1** [comment]
yeah i would think so. why is this decomposition being used? we shouldn't be using qiskit.transpile anywhere

**Comment by dowusu-antwi** [comment]
I don't think we are, but we've been using it in tomography since the qiskit tomography kit expects gates.

**Comment by dowusu-antwi** [comment]
I guess the question that @knsmith brought up is: should we somehow discourage users from using the qiskit transpiler on our custom gates?

**Comment by richrines1** [comment]
> I don't think we are, but we've been using it in tomography since the qiskit tomography kit expects gates.

ah ok, yeah those might not be super representative of the results we'd expect for circuits submitted to superstaq then

**Comment by richrines1** [comment]
> I guess the question that @knsmith brought up is: should we somehow discourage users from using the qiskit transpiler on our custom gates?

i guess i don't see when this would come up? if the user is instantiating the custom gates themselves, they probably intend to run a circuit with exactly those gates (in which case they wouldn't use `transpile`). otherwise, the custom gates are more there to express the compiled circuits (i.e. the user submits an arbitrary circuit, we compile it down using these custom gates, and submit that compiled circuit/return it to the user)

**Comment by richrines1** [comment]
(definitely not sure though - happy to meet quickly if you wanna discuss in more detail)

**Comment by knsmith** [comment]
I think the problem was that I hit a strange corner case that may not come up frequently. After placing a pulse gate in a circuit, I built the 12^n state prep/measurement circuits that contained gates like H, etc for tomography. However, I forgot to transpile again after they were built, and when the circuits reached the back end, all of the pulse gates would be transformed into their gate level approximation that's defined in SS. This happened even though I had added the custom calibration to the circuit.

Sounds like we just have to make our own tomography routines to avoid from this happening in the future :)
